Create update-cart (#6523)
umbraco/UmbracoDocs: 13/umbraco-commerce/how-to-guides/update-cart.md

---
description: How to update your cart.
---

# Update Card

Functionality is needed to update the cart once an item has been added. In this guide, you can learn how to add this functionality. 

> You need a new page for the user to give a summary of what is already in the cart and the ability to update each item.

Reviewer: Suggestion to shorten the sentence: 

```suggestion
You need a new page to summarize the items in the cart and allow users to update each item.
```

```csharp
public class CartSurfaceController(IUmbracoContextAccessor umbracoContextAccessor,
                                   IUmbracoDatabaseFactory databaseFactory, 
                                   ServiceContext services, AppCaches appCaches, 
                                   IProfilingLogger profilingLogger, 
                                   IPublishedUrlProvider publishedUrlProvider, 
                                   IUmbracoCommerceApi commerceApi) 
                                   : SurfaceController(umbracoContextAccessor, databaseFactory, services, appCaches, profilingLogger, publishedUrlProvider)
{
}
```

The `CartDto` is a class that passes data to the Controller. This is a class that has a property for the `productReference` and an array of `OrderLineQuantityDto[]`.

```csharp
    public class CartDto
    {
        public string ProductReference { get; set; }
        public OrderLineQuantityDto[] OrderLines { get; set; }
    }

    public class OrderLineQuantityDto
    {
        public Guid Id { get; set; }
        public decimal Quantity { get; set; }
    }
```

{% hint style="warning" %}

> `ProductReference` is not used here but is an example of how you could pass the product reference to the controller for other similar related tasks.

Reviewer: Suggestion to shorten the sentence:

```suggestion
The `ProductReference` is not used here, but it is an example of how to pass the product reference to the controller for similar tasks.
```
